temporian/core/data/duration.py

```diff
@@ -20,14 +20,48 @@
 """
 import datetime
 from enum import Enum
-from typing import Union
+from typing import Union, Optional
 
 import numpy as np
 
 # Unit for durations
-Duration = float
+NormalizedDuration = float
+Duration = Union[float, int]
 
 Timestamp = Union[np.datetime64, datetime.datetime, int, float]
+NormalizedTimestamp = float
+
+
+def normalize_duration(x: Duration) -> NormalizedDuration:
+    if isinstance(x, (int, float)) and x > 0:
+        return NormalizedDuration(x)
+
+    raise ValueError(
+        "A duration should be a strictly positive number of type float,"
+        f" int or tp.Duration. Got {x!r} of type {type(x)}."
+    )
+
+
+def normalize_timestamps(x: Timestamp) -> NormalizedTimestamp:
+    if isinstance(x, np.datetime64):
+        return x.astype("datetime64[ns]").astype(np.float64) / 1e9
+
+    if isinstance(x, datetime.datetime):
+        return float(x.timestamp())
+
+    if isinstance(x, int):
+        return float(x)
+
+    if isinstance(x, float):
+        return x
+
+    raise ValueError(f"Invalid timestamp {x!r} of type {type(x)}.")
+
+
+def normalize_timestamps_or_none(x: Optional[Timestamp]) -> NormalizedTimestamp:

```

Comment:
I think this will create an issue with the python annotation as the return type annotation cannot depend on the argument value. Every use of normalize_timestamp that does not check for None will raise an annotation error.